githubdelivery/src/main/java/io/dockstore/githubdelivery/GithubDeliveryS3Client.java

```diff
@@ -179,32 +179,38 @@ private void submitGitHubDeliveryEventsByHour(String prefix, WorkflowsApi workfl
     private void submitGitHubDeliveryEventsByKey(String key, WorkflowsApi workflowsApi) {
         String deliveryid = key.split("/")[2]; //since key is in YYYY-MM-DD/HH/deliveryid format
         try {
-            String body = getObject(key);
-            JsonObject jsonObject = GSON.fromJson(body, JsonObject.class);
-            if (jsonObject.get("action").getAsString().equals("added") || jsonObject.get("action").getAsString().equals("removed")) {
-                InstallationRepositoriesPayload payload = getGitHubInstallationRepositoriesPayloadByKey(body, key);
+            String obj = getObject(key);
+            JsonObject jsonObject = GSON.fromJson(obj, JsonObject.class);
+            JsonObject body = jsonObject.get("body").getAsJsonObject();
+            String bodyString = body.toString();
+            if ("installation_repositories".equals(jsonObject.get("eventType").getAsString())) {
+                InstallationRepositoriesPayload payload = getGitHubInstallationRepositoriesPayloadByKey(bodyString, key);
                 if (payload != null) {
                     workflowsApi.handleGitHubInstallation(payload, deliveryid);
                 } else {
                     logReadError(key);
                 }
 
-            } else  {
-                if (jsonObject.get("deleted").getAsBoolean()) {
-                    PushPayload payload = getGitHubPushPayloadByKey(body, key);
+            } else if ("push".equals(jsonObject.get("eventType").getAsString())) {
+                //push events
+                if (body.get("deleted").getAsBoolean()) {
+                    PushPayload payload = getGitHubPushPayloadByKey(bodyString, key);
                     if (payload != null) {
                         workflowsApi.handleGitHubBranchDeletion(payload.getRepository().getFullName(), payload.getSender().getLogin(), payload.getRef(), deliveryid, payload.getInstallation().getId());
                     } else {
                         logReadError(key);
                     }
                 } else {
-                    PushPayload payload = getGitHubPushPayloadByKey(body, key);
+                    PushPayload payload = getGitHubPushPayloadByKey(bodyString, key);
                     if (payload != null) {
                         workflowsApi.handleGitHubRelease(payload, deliveryid);
                     } else {
                         logReadError(key);
                     }
                 }
+            } else {
+                LOG.error("Invalid JSON format for event {}", key);
```

Comment:
Suggesting that the log message be something like `invalid eventType <eventType> for key <key>` so we know what event type was ignored